openhtf/io/output/output_to_file.py

```diff
@@ -0,0 +1,32 @@
+"""Output to file.
+
+Output modules inherit from OutputToFile and overload methods
+based on output file requirements.
+
+"""
+
+import os
+import tempfile
+
+
+class OutputToFile(object):
+  """Class for saving to file."""
+
+  def Save(self):
```

Comment:
see above for the longer version, but basically, you should implement `__call__` here, have it invoke a separate method on self for formatting data (make the default just use pickle), and have subclasses override that method rather than having to implement `__call__` themselves.

you might also pull out the filename_pattern handling into a separate method, so that `__call__` is really just a handful of calls to other methods on `self`.